apps/api-extractor/src/api/model/Deserializer.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { IApiItemJson, IApiItemOptions, ApiItem, ApiItemKind } from './ApiItem';
+import { ApiClass } from './ApiClass';
+import { ApiEntryPoint } from './ApiEntryPoint';
+import { ApiMethod } from './ApiMethod';
+import { ApiModel } from './ApiModel';
+import { ApiNamespace } from './ApiNamespace';
+import { ApiPackage } from './ApiPackage';
+import { ApiInterface } from './ApiInterface';
+import { ApiPropertySignature } from './ApiPropertySignature';
+import { ApiParameter } from './ApiParameter';
+import { ApiMethodSignature } from './ApiMethodSignature';
+import { ApiProperty } from './ApiProperty';
+
+export class Deserializer {
+  public static deserialize(jsonObject: IApiItemJson): ApiItem {
+    const options: Partial<IApiItemOptions> = { };
+
+    switch (jsonObject.kind) {
+      case ApiItemKind.Class:
+        ApiClass.onDeserializeInto(options, jsonObject);
+        return new ApiClass(options as any); // tslint:disable-line:no-any
```

Comment:
>any [](start = 39, length = 3)

Why is this cast necessary? You can just say `const options: IApiItemOptions = { } as IApiItemOptions;` on line 19.

Reply:
No, because each of the `ApiItem` subclass constructors wants different interface types, for example `IApiEntryPointOptions`, `IApiEnumMemberOptions`, `IApiMethodOptions`, etc.

But I can at least replace `any` with the explicit types

---
In reply to: [236471387](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236471387) [](ancestors = 236471387)